src/cli/cli_udp.cpp

```diff
@@ -172,17 +157,70 @@ otError UdpExample::ProcessSend(int argc, char *argv[])
         messageInfo.mPeerPort = static_cast<uint16_t>(value);
     }
 
+    if (argc == 2 || argc == 4)
+    {
+        int typePos = curArg++;
+
+        if (strcmp(argv[typePos], "-s") == 0)
+        {
+            unsigned long value;
+            payloadType = kTypeAutoSize;
+            SuccessOrExit(error = Interpreter::ParseUnsignedLong(argv[curArg], value));
+            payloadLength = static_cast<uint16_t>(value);
+        }
+        else if (strcmp(argv[typePos], "-x") == 0)
+        {
+            payloadLength = static_cast<uint16_t>(strlen(argv[curArg]));
+            payloadType   = kTypeHexString;
+        }
+        else if (strcmp(argv[typePos], "-t") == 0)
+        {
+            payloadType = kTypeText;
+        }
+    }
+
     message = otUdpNewMessage(mInterpreter.mInstance, NULL);
     VerifyOrExit(message != NULL, error = OT_ERROR_NO_BUFS);
 
-    if (autoGenPayload)
-    {
-        error = WriteCharToBuffer(message, static_cast<uint16_t>(autoGenPayloadLength));
-    }
-    else
+    switch (payloadType)
     {
+    case kTypeText:
         error = otMessageAppend(message, argv[curArg], static_cast<uint16_t>(strlen(argv[curArg])));
+        break;
+    case kTypeAutoSize:
+        error = WriteCharToBuffer(message, payloadLength);
+        break;
+    case kTypeHexString:
+    {
+        uint8_t     buf[50];
+        int8_t      bufLen;
+        const char *hexIndex         = argv[curArg];
+        uint16_t    remainHexLen     = static_cast<uint16_t>(strlen(argv[curArg]));
+        uint16_t    conversionHexLen = 0;
+
+        while (remainHexLen > 0)
```

Comment:
verify `remainHexLen` is even, otherwise error invalid argument?

Reply:
the design of existing Hex2Bin is to allow odd and prepend 0 if odd